documentation/design/security-metadata.md

```diff
@@ -187,14 +188,25 @@ First two presented option are not to be used. All the other options will likely
 * There are no global items today - this can be simulated by putting those to directory.props
 * Even seemingly innocent tasks with seemingly innocent logging can spill possibly sensitive data (e.g. think the RAR task, logging all the inputs, while those are just reference related info - those can contain paths that might already by itself be sensitive info). Related: [#8493](https://github.com/dotnet/msbuild/issues/8493) 
 * `MSBuild` task can pose a boundary for some context passing (e.g. properties/items).
+* Properties/items can occure dynamically after the initial processing of the script - e.g. [`CreateProperty task`](https://learn.microsoft.com/en-us/visualstudio/msbuild/createproperty-task). That should not be a problem, but we should keep it in mind (as additional entrypoint of external data into internal data holders).
 * Task authors and consumers are posibly different personas with disconected codebases. For this reason we want to support ability to indicate that task input/output is meant to be a secret. A user of the task should follow the contract and denote the data to be mounted to the task appropriately (otherwise a build warning/error will be issued).
 
 # Suggested Implementation
```

Comment:
We should build on the classifier interface. A string value presumably may be non-tokenized (a chunk of text) and may contain multiple sensitive tokens. If we don't return information on those individual tokens (for a string replacement), we can only redact the complete string value provided to us. Maybe something that returns a non-empty list of sensitive tokens that need to be replaced from the input.

~~~
public ISet<string> NeedsRedaction(string value)
~~~